### DISCUSSION TITLE
Modularized App model

### DISCUSSION THREAD

**Discussion by markus-li** [DustinBrett/daedalOS#182] [Ideas]
I initially wrote this as a reply to a rejected pull request. But think it may be worth discussing separately.
While I can understand your reasoning I'm still hoping you will change your mind on split public folders. Taking your analogy of how c:\ works I'd say that if public was split to Minimal and Default it would allow to "install" the "software" to C:\ and then build on that with whichever additional pieces one would want. 
As a way to convey my very strong wishes for a modularized App model now, rather than later, I created a sample branch [https://github.com/markus-li/daedalOS/tree/apps](https://github.com/markus-li/daedalOS/tree/apps). To be complete, this self-contained app model would require public to be split. An app could be maintained in a separate repo and just be included as a submodule into the components/app folder. As an example I have moved out BoxedWine and made it into a self-contained app. During a normal build all apps as defined in "appsEnabledDefaults" are included. A list of available Apps are added in a commented-out state to "appsEnabled" during the build. To enable custom Apps they just need to be uncommented in the non-version-controlled file "appsEnabled". All 100% self-contained.
The part in the app which is in "components/apps/BoxedWine/public" is not currently automatically included in "public", but would be in a full version of doing this, same way as it is done in this PR regarding a split public folder.
It may not be ideal, but nothing says that you can't evolve the model, even with breaking changes. As it is now, many changes are "breaking" in terms of keeping "3rd party" apps working flawlessly. I do have an app or two I'd love to put in a repo which could be included as a submodule in git.
It would even be possible to auto-merge in dependencies into package.json based on content in the manifest.json file present in each app.

I'm aware of issue [#37](https://github.com/DustinBrett/daedalOS/discussions/37) and [#133](https://github.com/DustinBrett/daedalOS/discussions/133) but made this separate due to it being a direct suggestion on how to start having an initial modularized app model which can evolve to a final model over time.

_Original post by @markus-li in https://github.com/DustinBrett/daedalOS/issues/178#issuecomment-1186422452_

**Comment by DustinBrett** [comment]
Thanks for opening this, I do indeed want to continue the discussion. I need to think on this more and it's not my top priority at the moment, but when I get into third party apps and making things more modular, I will indeed take whatever is discussed here as a basis for the work.
